engine/runtime/src/main/scala/org/enso/compiler/PackageRepository.scala

```diff
@@ -230,8 +256,66 @@ object PackageRepository {
         pkg            = pkg,
         isLibrary      = true
       )
+      pkg
     }.toEither.left.map { error => Error.PackageLoadingError(error.getMessage) }
 
+    /** @inheritdoc */
+    override def initialize(): Either[Error, Unit] = {
+      val unprocessedPackages =
+        loadedPackages.keySet
+          .diff(loadedComponents.keySet)
+          .flatMap(loadedPackages(_))
+      unprocessedPackages.foldLeft[Either[Error, Unit]](Right(())) {
+        (result, pkg) =>
+          for {
+            _ <- result
```

Comment:
```suggestion
        (accumulator, pkg) =>
          for {
            _ <- accumulator
```

Not sure, but maybe `accumulator` would be more accurate than the very generic `result`? Especially as this accumulator's only purpose is to thread-through any errors.